core/src/main/java/com/onelogin/saml2/util/Util.java

```diff
@@ -1026,6 +1063,26 @@ public static Boolean validateMetadataSign(Document doc, X509Certificate cert, S
      * @return True if the sign is valid, false otherwise.
      */
     private static Map<String,Object> getSignatureData(Node signNode, String alg) {
+		return getSignatureData(signNode, alg, false);
```

Comment:
Maybe a bit risky to have an implicit rejectDeprecatedAlg=false no? (the safer flag `true` could be accidentally ignored)

Reply:
I can't set it true, I need to maintain backward compatibility. But notice that in the settings file I added the rejection of deprecated alg as default.